app/Http/Controllers/TopicController.php

```diff
@@ -0,0 +1,114 @@
+        }
+
+        try {
+            $current_time = time();
+            $input = [
+                "topic_name" => $request->topic_name,
+                "namespace_id" => $request->namespace,
+                "submit_time" => $current_time,
+                "submitter_nick_id" => $request->nick_name,
+                "go_live_time" =>  $current_time,
+                "language" => 'English',
+                "note" => isset($request->note) ? $request->note : "",
+                "grace_period" => 1
+            ];
+            DB::beginTransaction();
+            $topic = Topic::create($input);
+            if ($topic) {
+                $topicInput = [
+                    "topic_num" => $topic->topic_num,
+                    "nick_name_id" => $request->nick_name,
+                    "delegate_nick_name_id" => 0,
+                    "start" =>  $current_time,
+                    "camp_num" => 1,
+                    "support_order" => 1,
+                ];
+                ## If topic is created then add default support to that topic ##
+                $support = Support::create($topicInput);
+
+                if (isset($request->namespace) && $request->namespace == 'other') {
+
+                    ## Create new namespace request ##
+                    $othernamespace = trim($request->create_namespace, '/');
+                    $namespace = new Namespaces();
+                    $namespace->parent_id = 0;
+                    $namespace->name = '/' . $othernamespace . '/';
+                    $namespace->save();
+
+                    ## update namespace id ##
+                    $topic->namespace_id = $namespace->id;
+                    $topic->update();
+                }
+                DB::commit();
+                try {
+                    $topicLive = Topic::getLiveTopic($topic->topic_num, $request->asof);
+                    $camp = Camp::getLiveCamp($topic->topic_num, 1, $request->asof);
+                    $historylink = Util::getTopicCampUrl($topic->topic_num, 1, $topicLive, $camp , time());
+                    $dataEmail = (object) [
+                        "type" => "topic",
+                        "link" => $historylink,
+                        "historylink" => env('APP_URL_FRONT_END') . '/topic/history/' . $topic->topic_num,
+                        "object" => $topic->topic_name . " / " . $camp->camp_name,
+                    ];
+                    Event::dispatch(new ThankToSubmitterMailEvent($request->user(), $dataEmail));
+                } catch (Throwable $e) {
+                    $status = 403;
+                    $message = $e->getMessage();
+                    return $this->resProvider->apiJsonResponse($status, $message, null, null);
+                }
+                $status = 200;
+                $message = trans('message.success.topic_created');
+                return $this->resProvider->apiJsonResponse($status, $message, null, null);
```

Comment:
Probably, we can move the return outside of if and else and get rid of one return statement